src/main/java/games/strategy/engine/framework/map/download/FileSizeWatcher.java

```diff
@@ -10,26 +10,25 @@
  * the file is polled in a new thread for its file size which is then passed to the
  * consumer.
  */
-public class FileSizeWatcher {
-
+final class FileSizeWatcher {
   private final File fileToWatch;
-  private final Consumer<Integer> progressListener;
-  private boolean stop = false;
+  private final Consumer<Long> progressListener;
+  private volatile boolean stop = false;
```

Comment:
Added `volatile` here because I noted that `stop` is accessed from multiple threads.  Probably not broken because the Java memory model is eventually consistent, but no harm in correctness.